docs/self-host.md

```diff
@@ -0,0 +1,88 @@
+# Self-Hosted App
+
+<!--consumer-badges start -->
+
+![node][node-badge]
+
+<!--consumer-badges end -->
+
+## Deploy a Self-Hosted Instance
+
+Multiple options exist for deploying a self-hosted instance:
+
+### Install `@repository-settings/app` as a dependency of your own app (recommended)
+
+This option offers you the most flexibility
+
+#### Depend on the package from npm
+
+```shell
+npm install @repository-settings/app
+```
+
+#### Example node.js app
+
+```js
+import {Server, Probot, ProbotOctokit} from 'probot';
+import app from '@repository-settings/app';
+
+async function start() {
+  const log = getLog();
+  const server = new Server({
+    Probot: Probot.defaults({
+      appId: process.env.APP_ID,
+      privateKey: process.env.PRIVATE_KEY,
+      secret: process.env.WEBHOOK_SECRET,
+      Octokit: ProbotOctokit.defaults({
+        baseUrl: process.env.GH_API
+      }),
+      log: log.child({
+        name: 'repository-settings'
+      })
+    }),
+    log: log.child({
+      name: 'server'
+    })
+  });
+
+  process.on('SIGTERM', async () => {
+    console.log('Received SIGTERM, stopping server!');
+
+    await server.stop();
+  });
+
+  await server.load(app);
+
+  await server.start();
+};
+
+start();
+```
```

Comment:
interested in whether you think there is a simpler example that would help folks understand the goal, or even a more generic probot doc to link to, but wanted to provide some level of guidance here

Reply:
I think if you don't need any custom functionality and you want a long-running process, I think there is no code necessary at all, you should be able to do `"start": "probot run node_modules/@repository-settings/app/index.js"`, or maybe `@repository-settings/app` should expos its own binary like `"start": "repository-settings"`?

If users need more customization, for example for a serverless environment, then we can point them to https://probot.github.io/docs/development/#run-probot-programmatically and maybe provide an example for a serverless app?

Reply:
i like the idea of including a few different variations from simplest to most customizability. 

> maybe `@repository-settings/app` should expos its own binary like `"start": "repository-settings"`?

i like this idea